src/chainstate/stacks/db/mod.rs

```diff
@@ -760,6 +785,92 @@ impl StacksChainState {
+                for schedule in initial_vesting_schedules {
+                    vesting_schedule_count = vesting_schedule_count + 1;
+                    let value = unlocks_per_blocks.get(&schedule.block_height).unwrap_or(&0);
+                    let index = value + 1;
+                    unlocks_per_blocks.insert(schedule.block_height, index);
+
+                    clarity_tx.connection().as_transaction(|clarity| {
+                        clarity
+                            .with_clarity_db(|db| {
+                                let key = TupleData::from_data(vec![
+                                    (
+                                        "stx-height".into(),
+                                        Value::UInt(schedule.block_height.into()),
+                                    ),
+                                    ("index".into(), Value::UInt(index.into())),
+                                ])
+                                .unwrap();
+
+                                let value = TupleData::from_data(vec![
+                                    ("owner".into(), Value::Principal(schedule.address.clone())),
+                                    ("metadata".into(), Value::buff_from_byte(0)),
+                                    ("unlock-ustx".into(), Value::UInt(schedule.amount.into())),
+                                ])
+                                .unwrap();
+                                db.insert_entry(
+                                    &lockup_contract_id,
+                                    "internal-locked-stx",
+                                    Value::Tuple(key),
+                                    Value::Tuple(value),
+                                )?;
+
+                                let key = TupleData::from_data(vec![(
+                                    "stx-height".into(),
+                                    Value::UInt(schedule.block_height.into()),
+                                )])
+                                .unwrap();
+                                let value = TupleData::from_data(vec![(
+                                    "total-unlocked".into(),
+                                    Value::UInt(index.into()),
+                                )])
+                                .unwrap();
+                                db.insert_entry(
+                                    &lockup_contract_id,
+                                    "unlocked-stx-per-block",
+                                    Value::Tuple(key),
+                                    Value::Tuple(value),
+                                )?;
+
+                                Ok(())
+                            })
+                            .unwrap();
+                    });
+                }
+                println!(
+                    "Done initializing chain with {} vesting schedules",
+                    vesting_schedule_count
+                );
```

Comment:
I think we need an extra step here: you should be summing all the vested tokens, and use this value for setting the balance of the lockup contract.

Reply:
We should also make sure that this amount of tokens is not being minted twice (from the balances + from this routine).

Reply:
Note for myself: the total sum of all eventual vested STX must be allocated to the principal of the lockup contract, because this is where the STX payouts will be transferred from (as opposed to vesting minting STX).

Reply:
Done!

Reply:
@lgalabru this may not be desirable -- this means that `total-liquid-ustx` will reflect all of the still-locked tokens, which in turn makes it more expensive to participate in PoX (even though these tokens aren't liquid).

Reply:
I think we should be minting the locked tokens on-the-fly, by intercepting calls to the `lockup` contract and doing the minting there (much like how we have special-case logic for PoX token locks/unlocks).

Reply:
@jcnelson can we let @zone117x merge this PR as is and let me address your feedback in https://github.com/blockstack/stacks-blockchain/pull/2109, already revisiting the lockup contract?

Reply:
@lgalabru Can you open an issue to describe the problem first?

Reply:
Issue created https://github.com/blockstack/stacks-blockchain/issues/2139

Reply:
@zone117x beat me to it with https://github.com/blockstack/stacks-blockchain/issues/2139, thanks!